Demo-Code.py

```diff
@@ -37,22 +60,41 @@ def randomize_balls(max_balls, min_diam, max_diam, bg_width, bg_height):
         tup_list.append(tup)
     return tup_list
 
-def create_ball(bg_width, bg_height, max_diam, min_diam):
+def create_ball(bg_width: int, bg_height: int, max_diam: int, min_diam: int):
+    """
+    :param min_diam: min diameter of each ball
+    :param max_diam: max diameter of each ball
+    :param bg_width: width of the background
+    :param bg_height: height of the background
+    :return: a tuple containing the randomized top left corner coordinates and the randomized diameter of the ball
+    """
     diam = random.randint(min_diam, max_diam)
     x = random.randint(0, bg_width - (diam + 1))
     y = random.randint(0, bg_height - (diam + 1))
     tup = (x, y, diam)
     return tup
 
-def is_ball_legal(tup_list, tup):
+def is_ball_legal(tup_list: list, tup: tuple):
+    """
+    :param tup_list: list of all balls olready created
+    :param tup: a new ball
+    :return: true if the ball does not interrupt other balls, else false
+    """
     for i in tup_list:
         if (((i[0] + i[2] / 2.0) - (tup[0] + tup[2] / 2.0)) * ((i[0] + i[2] / 2.0) - (tup[0] + tup[2] / 2.0)) +
             ((i[1] + i[2] / 2.0) - (tup[1] + tup[2] / 2.0)) * ((i[1] + i[2] / 2.0) - (tup[1] + tup[2] / 2.0)) <=
             ((tup[2] + i[2]) / 2.0 + 1) * ((tup[2] + i[2]) / 2.0 + 1)):
             return False
     return True
     
-def circle_copy_stupid(ball_arr, bg_arr, x_top_left, y_top_left, diameter):  # diameter equals to width
+def circle_copy_stupid(ball_arr: np.ndarray, bg_arr: np.ndarray, x_top_left: int, y_top_left: int, diameter: int):  # diameter equals to width
+    """
+    :param ball_arr: numpy array representing the background picture
+    :param bg_arr: numpy array representing a ball picture
+    :param x_top_left: the top left x coordinate of the location the ball is pasted on the background
+    :param y_top_left: the top left y coordinate of the location the ball is pasted on the background
+    :param diameter: the diameter of the ball
+    """
     r = diameter / 2.0
     x_center, y_center = int(r - 0.25) + 0.5, int(r - 0.25) + 0.5
     for x_counter in range(diameter + 1):
```

Comment:
I promised Ittai that we will use his more efficient version of the function, please contact him and integrate it.

Reply:
I will wait for him to post his function on git

Reply:
I'll talk to him